client/consensus/manual-seal/src/lib.rs

```diff
@@ -171,286 +184,29 @@ pub async fn run_instant_seal<B, CB, E, C, A, SC, T>(
 	// into the transaction pool.
 	let commands_stream = pool.validated_pool()
 		.import_notification_stream()
-		.map(|_| {
+		.map(move |_| {
 			EngineCommand::SealNewBlock {
 				create_empty: false,
-				finalize: false,
+				finalize,
 				parent_hash: None,
 				sender: None,
 			}
 		});
 
+	let stream: Box<dyn Stream<Item=EngineCommand<<B as BlockT>::Hash>> + Unpin + Send> =
```

Comment:
Does it really require the full type here?

Reply:
No. I've reduced it to `Box<dyn Stream<Item = _> + Unpin>`. I don't think it can be reduced further. This line may become obsolete after we implement your other changes anyway.